llvm/lib/Target/EraVM/EraVMPostCodegenPrepare.cpp

```diff
@@ -0,0 +1,128 @@
+//===-- EraVMPostCodegenPrepare.cpp - EraVM Post CodeGenPrepare -*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This pass performs optimizations after CodeGenPrepare pass. Currently, it is
+// only performing transformation from signed to unsigned compares where it is
+// feasible.
```

Comment:
Could we refine this statement? We only do the transformation for `<s`, `>s` and constant in RHS. We probably don't care about `s>=, `s<=` as we expect LLVM to convert them to strict comparisons. So ultimately feasibility == const in RHS.

Reply:
Will update comment. We don't need to worry about ule and uge, since these will be canonicalized to ult and ugt as it is shown [here](https://godbolt.org/z/v1z6cxzf4).